pvlib/tests/test_shading.py

```diff
@@ -104,3 +107,87 @@ def test_sky_diffuse_passias_scalar(average_masking_angle, shading_loss):
     for angle, loss in zip(average_masking_angle, shading_loss):
         actual_loss = shading.sky_diffuse_passias(angle)
         assert np.isclose(loss, actual_loss)
+
+
+@pytest.fixture
+def true_tracking_angle_and_inputs():
+    # data retrieved from NREL Slope-Aware Backtracking for Single-Axis
+    # Trackers
+    # doi.org/10.2172/1660126 ; Accessed on 2023-11-06.
+    tzinfo = timezone(timedelta(hours=-5))
+    axis_tilt_angle = 9.666  # deg
+    axis_azimuth_angle = 195.0  # deg
+    timedata = pd.DataFrame(
+        columns=("Apparent Elevation", "Solar Azimuth", "True-Tracking"),
+        data=(
+            (2.404287, 122.791770, -84.440),
+            (11.263058, 133.288729, -72.604),
+            (18.733558, 145.285552, -59.861),
+            (24.109076, 158.939435, -45.578),
+            (26.810735, 173.931802, -28.764),
+            (26.482495, 189.371536, -8.475),
+            (23.170447, 204.136810, 15.120),
+            (17.296785, 217.446538, 39.562),
+            (9.461862, 229.102218, 61.587),
+            (0.524817, 239.330401, 79.530),
+        ),
+    )
+    timedata.index = pd.date_range(
+        "2019-01-01T08", "2019-01-01T17", freq="1H", tz=tzinfo
+    )
+    timedata["Apparent Zenith"] = 90.0 - timedata["Apparent Elevation"]
+    return (axis_tilt_angle, axis_azimuth_angle, timedata)
+
+
+def test_projected_solar_zenith_angle_numeric(true_tracking_angle_and_inputs):
+    psz_func = shading.projected_solar_zenith_angle
+    axis_tilt, axis_azimuth, timedata = true_tracking_angle_and_inputs
+    psz = psz_func(
+        axis_tilt,
+        axis_azimuth,
+        timedata["Apparent Elevation"],
+        timedata["Solar Azimuth"],
+    )
+    assert_allclose(psz, timedata["True-Tracking"], atol=1e-3)
+    # test equivalence against pvlib.tracking.singleaxis
+    singleaxis = pvlib.tracking.singleaxis(timedata["Apparent Zenith"],
+                                           timedata["Solar Azimuth"],
+                                           axis_tilt, axis_azimuth,
+                                           backtrack=False)
+    assert_allclose(psz, singleaxis["tracker_theta"])
```

Comment:
It's true that `singleaxis` currently implements the same calculation, but the idea is to follow this PR with another that has `singleaxis` call this new function instead of implementing the same calculations.  So testing against a call to `singleaxis` makes sense right now, but pretty soon it won't make sense (it will essentially just test this new function against itself).

I think what we should do is use `singleaxis` to calculate some test values today, and hardcode those values into these tests.

Reply:
Ah! That makes sense, I feel stupid for not copying that implementation from the start.

Reply:
Wouldn't it be better to just copy it? I mean, it already works and is commented.

Reply:
The code seems essentially the same between the two implementations, so I don't think it matters.  The new one might be slightly easier to trace to the reference.  The comments would be nice to copy over though (assuming they are still correct!)